src/renderer/gdi/paint.cpp

```diff
@@ -347,6 +350,7 @@ using namespace Microsoft::Console::Render;
 
             const auto text = cluster.GetText();
             polyString += text;
+            polyString.back() &= softFontCharMask;

```

Comment:
I wonder -- this may impact surrogate pairs negatively. We don't _fully_ support them here, but they do somewhat operate acceptably. I was wondering whether somebody setting the soft font and then emitting a surrogate pair would be a rare/freak/unlikely occurrence, but then I remembered that they can be paired with normal text (right?)

Reply:
I was a little worried about this too, but I've double checked the logic and I think it's OK. The renderer splits soft font runs from regular font runs, so when it's in a soft font run there should never be a surrogate pair, so the cluster text will always be a single wchar_t. And when it's in a regular font run the mask has all bits sets, so although it may only be masking half of the surrogate pair, it's essentially a noop, so it doesn't matter.

I've also run some tests mixing soft fonts with emojis, and they seem to work fine.

Reply:
Absolutely awesome. Thanks.